presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -552,6 +558,13 @@ public boolean transitionToFinishing()
             return false;
         }
 
+        try {
+            metadata.endQuery(session);
```

Comment:
But if we the `endQuery` failure is ignored it could be potential hidden resource leak. So, from my prespective, it is better to fail the query explicitly.